### PR TITLE
Fix crazyflie web component

### PR DESCRIPTION
**Pull request by ThomasOliverKimble**
**Description**
- Fix the new Crazyflie's web component view. (see [here](https://cyberbotics.com/doc/guide/crazyflie?version=fix-crazyflie-web-component))
- Update Crazyflie world by modifying viewpoint and adding a `follow` to the `Viewpoint`

crazyflie_demo

### CODE REVIEW

**Review by knmcguire — CHANGES_REQUESTED** [review]
Just a tiny change necessary here :) but it looks good at https://www.cyberbotics.com/doc/guide/crazyflie?version=fix-crazyflie-web-component. It was only hanging in the air.

**Comment by knmcguire** [comment]
I guess after we change the json file, it needs to be run again right?

**Comment by ThomasOliverKimble** [comment]
> Just a comment. Concerning the meshes in the x3d, it should be possible to re-use the existing ones by setting the url to: `webots://path/to/meshes` (it's what we do for the panda for instance, but I'm not sure we do it for all). I think you can delete them and fix the urls manually in this PR, I'll open an issue about it so that we can improve the web component script.

I have just done the modifications, but yes we should look into this.

File: docs/guide/scenes/crazyflie/crazyflie.x3d [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
...nce/mesh' url='"webots://projects/robots/bitcraze/crazyflie/protos/meshes/cw_prop.dae"'></Mesh></Shape></Transform></Group><Group type='propeller'><Transform id='n37' docUrl='https://cyberbotics.com/doc/reference/solid' translation='-0.031 0.031 0.008' name='solid' type='solid'><Shape id='n38' castShadows='true'><PBRAppearance id='n39' docUrl='https://cyberbotics.com/doc/reference/pbrappearance' USE='n27'/><Mesh id='n40' docUrl='https://cyberbotics.com/doc/reference/mesh' url='"webots://projects/robots/bitcraze/crazyflie/protos/meshes/ccw_prop.dae"'></Mesh></Shape></Transform></Group><Group type='propeller'><Transform id='n43' docUrl='https://cyberbotics.com/doc/reference/solid' translation='0.031 0.031 0.007' name='solid' type='solid'><Shape id='n44' castShadows='true'><PBRAppearance id='n45' docUrl='https://cyberbotics.com/doc/reference/pbrappearance' USE='n27'/><Mesh id='n46' docUrl='https://cyberbotics.com/doc/reference/mesh' url='"webots://projects/robots/bitcraze/crazyflie/protos/meshes/cw_prop.dae"'></Mesh></Shape></Transform></Group><Cylinder id='n47' docUrl='https://cyberbotics.com/doc/reference/cylinder' height='0.03' radius='0.05' subdivision='36'></Cylinder></Transform>
```
Comment (knmcguire) [inline docs/guide/scenes/crazyflie/crazyflie.x3d:15]: This is one of the generated files right? it seems it still has the old value

```suggestion
<Transform id='n10' docUrl='https://cyberbotics.com/doc/guide/crazyflie' translation='0 0 0.015' name='crazyflie' controller="crazyflie" type='robot'><Transform id='n11' docUrl='https://cyberbotics.com/doc/reference/transform' translation='0 0 -0.015'><CadShape id='n13' docUrl='https://cyberbotics.com/doc/reference/cadshape' url='"webots://projects/robots/bitcraze/crazyflie/protos/meshes/cf2_assembly.dae"'></CadShape></Transform><Transform id='n14' docUrl='https://cyberbotics.com/doc/reference/transform' translation='0 0 0.02'></Transform><Group type='propeller'><Transform id='n25' docUrl='https://cyberbotics.com/doc/reference/solid' translation='0.031 -0.031 0.008' name='solid' type='solid'><Shape id='n26' castShadows='true'><PBRAppearance id='n27' docUrl='https://cyberbotics.com/doc/reference/pbrappearance' baseColor='0 0 0' metalness='0.3'></PBRAppearance><Mesh id='n28' docUrl='https://cyberbotics.com/doc/reference/mesh' url='"webots://projects/robots/bitcraze/crazyflie/protos/meshes/ccw_prop.dae"'></Mesh></Shape></Transform></Group><Group type='propeller'><Transform id='n31' docUrl='https://cyberbotics.com/doc/reference/solid' translation='-0.031 -0.031 0.008' name='solid' type='solid'><Shape id='n32' castShadows='true'><PBRAppearance id='n33' docUrl='https://cyberbotics.com/doc/reference/pbrappearance' USE='n27'/><Mesh id='n34' docUrl='https://cyberbotics.com/doc/reference/mesh' url='"webots://projects/robots/bitcraze/crazyflie/protos/meshes/cw_prop.dae"'></Mesh></Shape></Transform></Group><Group type='propeller'><Transform id='n37' docUrl='https://cyberbotics.com/doc/reference/solid' translation='-0.031 0.031 0.008' name='solid' type='solid'><Shape id='n38' castShadows='true'><PBRAppearance id='n39' docUrl='https://cyberbotics.com/doc/reference/pbrappearance' USE='n27'/><Mesh id='n40' docUrl='https://cyberbotics.com/doc/reference/mesh' url='"webots://projects/robots/bitcraze/crazyflie/protos/meshes/ccw_prop.dae"'></Mesh></Shape></Transform></Group><Group type='propeller'><Transform id='n43' docUrl='https://cyberbotics.com/doc/reference/solid' translation='0.031 0.031 0.007' name='solid' type='solid'><Shape id='n44' castShadows='true'><PBRAppearance id='n45' docUrl='https://cyberbotics.com/doc/reference/pbrappearance' USE='n27'/><Mesh id='n46' docUrl='https://cyberbotics.com/doc/reference/mesh' url='"webots://projects/robots/bitcraze/crazyflie/protos/meshes/cw_prop.dae"'></Mesh></Shape></Transform></Group><Cylinder id='n47' docUrl='https://cyberbotics.com/doc/reference/cylinder' height='0.03' radius='0.05' subdivision='36'></Cylinder></Transform>
```

**Review by ad-daniel — DISMISSED** [review]
Looks good, thank you!

File: docs/guide/scenes/crazyflie/crazyflie.x3d (hunk 1 above)
Comment (knmcguire) [reply docs/guide/scenes/crazyflie/crazyflie.x3d:15]: I mean in the z-axis: translation='0 0 0.05'  should be 0 0 0.015.

**Review by knmcguire — CHANGES_REQUESTED** [review]

File: docs/guide/scenes/crazyflie/crazyflie.x3d (hunk 1 above)
Comment (ad-daniel) [reply docs/guide/scenes/crazyflie/crazyflie.x3d]: It depends, you want the default translation to be `0 0 0.015` when the PROTO is inserted in the world as well (in this case it requires changing the PROTO) ? Or just have a non-default pose just for the web component (in which case the json needs to be changed)?

If the PROTO needs to be modified, then the documentation page need to be updated accordingly (as it's generated from the PROTO)

Comment (ThomasOliverKimble) [reply docs/guide/scenes/crazyflie/crazyflie.x3d]: Indeed, changing the json file for the web component viewer won't change the default-pose of the proto (here 0 0 0.015).

I have modified the height to the default pose, but I personally think it looks better floating in the web component viewer (see comparison below)




let me know!

Comment (knmcguire) [reply docs/guide/scenes/crazyflie/crazyflie.x3d]: yes this looks good!

**Review by knmcguire — APPROVED** [review]

**Comment by ThomasOliverKimble** [comment]
> Seems to not float anymore. https://www.cyberbotics.com/doc/guide/crazyflie?version=fix-crazyflie-web-component

Yes, if you want it to float, the translation has to be: '0 0 0.05' in the json/x3d files.

**Comment by knmcguire** [comment]
That is fine :) the props are not turning so it actually looks a bit weird if it is floating. I prefer it being on the ground for now.

**Review by ad-daniel — APPROVED** [review]
Thank you

**Comment by knmcguire** [comment]
thanks for handling this @ThomasOliverKimble !

**Pull request merged**